dev/RadialGradientBrush/RadialGradientBrush.cpp

```diff
@@ -0,0 +1,241 @@
+        }
+        else if (property == s_MappingModeProperty)
+        {
+            UpdateCompositionGradientMappingMode();
+        }
+        else if (property == s_InterpolationSpaceProperty)
+        {
+            UpdateCompositionInterpolationSpace();
+        }
+    }
+}
+
+void RadialGradientBrush::OnFallbackColorChanged(const winrt::DependencyObject& sender, const winrt::DependencyProperty& args)
+{
+    UpdateFallbackBrush();
+}
+
+void RadialGradientBrush::EnsureCompositionBrush()
+{
+    if (m_isConnected && !m_brush)
+    {
+        auto compositor = winrt::Window::Current().Compositor();
+
+        if (SharedHelpers::IsCompositionRadialGradientBrushAvailable())
+        {
+            // If CompositionRadialGradientBrush is available then use it to render a gradient.
+            m_brush = compositor.CreateRadialGradientBrush();
+
+            UpdateCompositionGradientEllipseCenter();
+            UpdateCompositionGradientEllipseRadius();
+            UpdateCompositionGradientOriginOffset();
+            UpdateCompositionGradientStops();
+            UpdateCompositionGradientMappingMode();
+            UpdateCompositionInterpolationSpace();
+        }
+        else
+        {
+            // If CompositionRadialGradientBrush isn't available then render using the FallbackColor.
+            m_brush = compositor.CreateColorBrush();
+            UpdateFallbackBrush();
+        }
+
+        CompositionBrush(m_brush);
+    }
+}
+
+void RadialGradientBrush::UpdateCompositionGradientEllipseCenter()
+{
+    MUX_ASSERT(SharedHelpers::IsCompositionRadialGradientBrushAvailable());
+
+    auto compositionGradientBrush = m_brush.try_as<winrt::CompositionRadialGradientBrush>();
+
+    if (compositionGradientBrush)
+    {
+        compositionGradientBrush.EllipseCenter(winrt::float2(EllipseCenter().X, EllipseCenter().Y));
+    }
+}
+
+void RadialGradientBrush::UpdateCompositionGradientEllipseRadius()
+{
+    MUX_ASSERT(SharedHelpers::IsCompositionRadialGradientBrushAvailable());
+
+    auto compositionGradientBrush = m_brush.try_as<winrt::CompositionRadialGradientBrush>();
+
+    if (compositionGradientBrush)
+    {
+        compositionGradientBrush.EllipseRadius(winrt::float2(EllipseRadius().X, EllipseRadius().Y));
+    }
+}
+
+void RadialGradientBrush::UpdateCompositionGradientMappingMode()
+{
+    MUX_ASSERT(SharedHelpers::IsCompositionRadialGradientBrushAvailable());
+
+    auto compositionGradientBrush = m_brush.try_as<winrt::CompositionRadialGradientBrush>();
+
+    if (compositionGradientBrush)
+    {
+        switch (MappingMode())
+        {
+        case winrt::BrushMappingMode::Absolute:
+            compositionGradientBrush.MappingMode(winrt::Windows::UI::Composition::CompositionMappingMode::Absolute);
+            break;
+
+        case winrt::BrushMappingMode::RelativeToBoundingBox:
+            compositionGradientBrush.MappingMode(winrt::Windows::UI::Composition::CompositionMappingMode::Relative);
+            break;
+        }
```

Comment:
Are these the only two modes? should we throw if a different value is provided ?

Reply:
I used the existing [BrushMappingMode](https://docs.microsoft.com/uwp/api/windows.ui.xaml.media.brushmappingmode) enum which has only ever had two values for ~20 years and I don't expect it will be expanded.

Is it convention to guard against future potential additions to an enum and throw?

We could also make `Relative` the default instead of an explicit case, but that's already the behavior we'd get since if we don't explicitly set any mapping mode then the brush will just the composition layer's existing default (Relative).

Reply:
I'm ok with falling back to a default behavior instead of throwing, but vote for making it explicit in the code with a default tag for the switch statement.

Reply:
+1
While it is not explicitly called out as a rule in the C++ guidelines, es79(https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#es79-use-default-to-handle-common-cases-only) and Enum2's (https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#enum2-use-enumerations-to-represent-sets-of-related-named-constants) enforcement rules imply that call switch statement should have a default case, even if it is a do nothing, or a throw.

Reply:
Added default case and made the RelativeToBoundingBox case `[[fallthrough]]` - let me know if you'd prefer explicit duplication instead